tests/unit/parser.js

```diff
@@ -33,6 +34,83 @@ tap.test('Parser', {autoend: true}, t => {
         }
       }
     )
+
+    t.test(
+      'should throw error when inputFormat for endpoint does not match the request body content type',
+      async t => {
+        t.plan(1)
+        const ctx = {
+          state: {
+            uuid: 'randomUidForRequest',
+            metaData: {
+              name: 'Testing endpoint'
+            }
+          },
+          get: name => {
+            const contentHeader = 'application/unsupported'
+            if (name) {
+              return contentHeader
+            }
+          }
+        }
+        const inputFormat = 'JSON'
+
+        try {
+          await parseIncomingBody(ctx, inputFormat)
+        } catch (error) {
+          t.equal(
+            error.message
+              .match(
+                /Supplied input format does not match incoming content-type:/
```

Comment:
@bradsawadye Having to match the error message like this is a bit weird. Please could you remove the Endpoint Name and UUID from the error object. The error object should not contain those details as we should include this in the logger.

```js
      throw new Error(
        `Supplied input format does not match incoming content-type: Expected ${inputFormat.toLowerCase()} format, but received ${
          ctx.get('Content-Type').split('/')[1]
        }`
      )
```

After changing the error message you should be able to match the error.message directly